service/web/server/docs/openapi.yaml

```diff
@@ -0,0 +1,623 @@
+
+  /forecast/{testid}:
+    parameters:
+      - $ref: '#/components/parameters/TestId'
+    put:
+      summary: Submit a forecast query
+      tags:
+        - data
+      description: Sends a ForecastQuery payload and returns a TimeSeries response.
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ForecastQuery'
+      responses:
+        '200':
+          description: Successful response with forecast time series.
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  message:
+                    type: string
+                    example: "Forecast query processed successfully."
+                  payload:
+                    $ref: '#/components/schemas/TimeSeries'
+                  status:
+                    type: integer
+                    example: 200
+  /results/{testid}:
+    parameters:
+      - $ref: '#/components/parameters/TestId'
+    put:
+      summary: Submit a results query
+      tags:
+        - data
+      description: Sends a ResultsQuery payload and returns a TimeSeries response.
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ResultsQuery'
+      responses:
+        '200':
+          description: Successful response with results time series.
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  message:
+                    type: string
+                    example: "Results query processed successfully."
+                  payload:
+                    $ref: '#/components/schemas/TimeSeries'
+                  status:
+                    type: integer
+                    example: 200
+  /kpi/{testid}:
+    parameters:
+      - $ref: '#/components/parameters/TestId'
+    get:
+      summary: Get KPIs for a test
+      tags:
+        - data
+      description: Returns key performance indicators for the running test.
+      responses:
+        '200':
+          description: Successful response with KPIs.
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  message:
+                    type: string
+                    example: "KPIs retrieved successfully."
+                  payload:
+                    type: object
+                    additionalProperties:
+                      type: number
+                    description: Mapping of KPI name to numeric value.
+                  status:
+                    type: integer
+                    example: 200
+
+  /advance/{testid}:
+    parameters:
+      - $ref: '#/components/parameters/TestId'
+    post:
+      summary: Advance the simulation
```

Comment:
Suggest "Advance the simulation on control step with optional control input overrides."